vehicle/lib/Definitions.vcl

```diff
@@ -44,6 +44,15 @@ geRatTensorReduced xs ys = reduceAnd True (xs >=. ys)
 gtRatTensorReduced : Tensor Real (dim :: dims) -> Tensor Real (dim :: dims) -> Bool
 gtRatTensorReduced xs ys = reduceAnd True (xs >. ys)
 
+--------------------------------------------------------------------------------
+-- TensorLike
+--------------------------------------------------------------------------------
+
+record TensorLike r t dims {{isTensorType : IsTensorType t dims}} where
+  { toTensorLike     : r -> Tensor t dims
```

Comment:
I would probably call this `toTensor` and `fromTensor`. The current names suggest that the types should be other way around, i..e your converting to `r`?